docs/source/usage.rst

```diff
@@ -24,6 +24,97 @@ To run MovieBot with default configuration execute the command below.
+
+   RECOMMENDER: "slot_based"
+
+   TELEGRAM: False # execute the code on Telegram
+
+   POLLING: False # True when using Telegram without server
+
+   BOT_TOKEN_PATH: config/bot_token.yaml
+
+   BOT_HISTORY: # save the conversational logs for Telegram users
+      save: True
+      path: conversation_history
+
+   DEBUG: False
+
+Flask REST API
+""""""""""""""
+
+To run MovieBot with Flask REST API, you need to add the following line to the configuration file: `FLASK_REST: True`.
+
+By default, MovieBot will be running locally on the port 5001. To modify the host or the port, you need to modify the file `moviebot/controller/server_rest.py`.
+
+We do not provide a web client to interact with MovieBot. The communication is performed through HTTP requests.
+The following example shows how to start a conversation with the MovieBot using Python:
+
+.. code-block:: python
+
+   import requests
+
+   uri = "http://127.0.0.1:5001"
+
+   # Start a new conversation
+   r = requests.post(
+            uri,
+            json={
+                     "message": {"text": "/start"},
+                     "sender": {"id": "test_user"}, 
+            },
+        )
+   print(r.json())
+   >>> {'message': {'attachments': [], 'intent': '', 'text': 'Hi there. I am IAI MovieBot, your movie recommending buddy. I can recommend you movies based on your preferences.\nI will ask you a few questions and based on your answers, I will try to find a movie for you.\n\n\n\nINSTRUCTIONS\n\nTo start the conversation say Hi or Hello, or simply enter you preferences ("I want a horror movie from the 90s").\n\nTo restart the recommendation process, issue "/restart".\n\nTo end the conversation, issue "/exit" or say Bye/Goodbye.\n\nTo see these instructions again, issue: "/help".'}, 'recipient': {'id': 'test_user'}}
+
+Please note that in this configuration, the first message must be a `/start` message.
+
+Flask SocketIO
+""""""""""""""
+
+To run MovieBot with Flask SocketIO, you need to add the following line to the configuration file: `FLASK_SOCKET: True`.
+
+By default, MovieBot will be running locally on the port 5000. To modify the host or the port, you need to modify the file `moviebot/controller/server_socket.py`.
+
+We provide a simple `web widget <https://www.npmjs.com/package/iaigroup-chatwidget>`_ to interact with the MovieBot. It can be added to any web page by adding the following lines to the associated HTML code:
+
+.. code-block:: HTML
+
+   <script type="text/javascript"
+        src="https://cdn.jsdelivr.net/npm/iaigroup-chatwidget@1.1.2/build/bundle.min.js"></script>
+    <script type="text/javascript">
+        ChatWidget({
+            name: "MovieBot",
+            serverUrl: "http://127.0.0.1:5001",
+            socketioPath: "/socket.io/",
+            useFeedback: false,
+            useLogin: false,
+        });
+    </script>
+
+A button to discuss with MovieBot should be placed at the bottom right of your web page.
```

Comment:
Nit: discuss seems a bit strange in this context.

Reply:
Replaced by interact.